walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,100 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
 
+
+/*  Find time slot(s) that all the required attendees are availble
+    Include optional attendees if at least one time slot can be found
+    Otherwise return the time slots that fit just the mandatory attendees.
+*/
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Initialize a time slot (the whole day)
+    List<TimeRange> timeSlots = new ArrayList<>();
+    timeSlots.add(TimeRange.fromStartDuration(0, 24 * 60));
+
+    // Find time slots for required attendees
+    timeSlots = queryForEvent(events, timeSlots, request.getAttendees());
+    timeSlots = deleteUnqualifiedQuery(timeSlots, request.getDuration());
+
+    // Save the result in timeSlotsForRequired
+    // in case no time slot can be found when optional attendees are included
+    List<TimeRange> timeSlotsForRequired = new ArrayList<TimeRange>(timeSlots);
+
+    // Check for the availability of optional attendees
+    timeSlots = queryForEvent(events, timeSlots, request.getOptionalAttendees());
```

Comment:
Does "timeSlots" include slots for required attendees, or only optional attendees?

Reply:
The variable timeSlots is first modified to fit all the required attendees' schedules, and the result is saved in timeSlotsForRequired. Then it goes on to check all the optional attendees, so after line 43, timeSlots include slots for both required and optional attendees. Do you think it is better to change the variable name to something like timeSlotsForAll? (I think the only problem with this is that making a copy of timeSlotsForAll and naming it timeSlotsForRequired does not make a lot of sense) Or do you think it is better to just add some comments here?